### PR TITLE
IA-1593: Stocks: replace the icons "settings" with an icon "edit"

### PR DESCRIPTION
**Pull request by hakifran**
Explain what problem this PR is resolving
- Stocks: replace the icons "settings" with an icon "edit"
Related JIRA tickets : [POLIO-1593](https://bluesquare.atlassian.net/browse/POLIO-1593)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes
- replace the icons "settings" with an icon "edit" in stock management

## How to test
- Go in Polio -> Vaccine module -> stock management -> stock variation
- Check Form A, Destruction report and incident report and check the edit ico

## Print screen / video
[Screencast from 2024-07-08 11-21-12.webm](https://github.com/BLSQ/iaso/assets/19631540/ea6bd5c3-36f9-472c-b3c2-69b974d6182a)

[POLIO-1593]: https://bluesquare.atlassian.net/browse/POLIO-1593?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hakifran** [comment]
@cypress

File: hat/assets/js/apps/Iaso/components/Buttons/EditIconButton.tsx [hunk 1]
```diff
@@ -13,17 +13,20 @@ type Props = {
...
     message?: IntlMessage;
     disabled?: boolean;
+    overridedIcon?: any;
 };
 
 export const EditIconButton: FunctionComponent<Props> = ({
     onClick,
     message,
     disabled = false,
+    overridedIcon,
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/components/Buttons/EditIconButton.tsx]: overridedIcon --> overrideIcon ( to stay consistent)
```suggestion
    overrideIcon,
```

File: hat/assets/js/apps/Iaso/components/Buttons/EditIconButton.tsx [hunk 2]
```diff
@@ -13,17 +13,20 @@ type Props = {
...
 }) => {
     return (
         <IconButton
             onClick={onClick}
             icon="edit"
+            overrideIcon={overridedIcon}
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/components/Buttons/EditIconButton.tsx]:
```suggestion
            overrideIcon={overrideIcon}
```

File: plugins/polio/js/src/domains/VaccineModule/StockManagement/StockVariation/Table/columns.tsx [hunk 3]
```diff
@@ -98,7 +99,7 @@ export const useFormATableColumns = (
                             <EditFormA
                                 id={settings.row.original.id}
                                 formA={settings.row.original}
-                                iconProps={{}}
+                                iconProps={{ overridedIcon: EditIcon }}
```
Comment (quang-le) [inline plugins/polio/js/src/domains/VaccineModule/StockManagement/StockVariation/Table/columns.tsx]:
```suggestion
                                iconProps={{ overrideIcon: EditIcon }}
```

File: plugins/polio/js/src/domains/VaccineModule/StockManagement/StockVariation/Table/columns.tsx [hunk 4]
```diff
@@ -176,7 +177,7 @@ export const useDestructionTableColumns = (
                             <EditDestruction
                                 id={settings.row.original.id}
                                 destruction={settings.row.original}
-                                iconProps={{}}
+                                iconProps={{ overridedIcon: EditIcon }}
```
Comment (quang-le) [inline plugins/polio/js/src/domains/VaccineModule/StockManagement/StockVariation/Table/columns.tsx]:
```suggestion
                                iconProps={{ overrideIcon: EditIcon }}
```

File: plugins/polio/js/src/domains/VaccineModule/StockManagement/StockVariation/Table/columns.tsx [hunk 5]
```diff
@@ -264,7 +265,7 @@ export const useIncidentTableColumns = (
                             <EditIncident
                                 id={settings.row.original.id}
                                 incident={settings.row.original}
-                                iconProps={{}}
+                                iconProps={{ overridedIcon: EditIcon }}
```
Comment (quang-le) [inline plugins/polio/js/src/domains/VaccineModule/StockManagement/StockVariation/Table/columns.tsx]:
```suggestion
                                iconProps={{ overrideIcon: EditIcon }}
```

**Review by quang-le — DISMISSED** [review]
See suggestions. Just typos really. I still think that if we change the icon in stock, we should also change it in supply chain, as the 2 modules are quite interconnected

**Comment by hakifran** [comment]
> See suggestions. Just typos really. I still think that if we change the icon in stock, we should also change it in supply chain, as the 2 modules are quite interconnected

@quang-le the edit ico has been added to supply chain too

**Review by butofleury — APPROVED** [review]

**Pull request merged**